app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -636,6 +636,79 @@ public void onTestNewFunction() {
     }
 
 
+    private void showIncomingCall(String name, String number){
+        LOG.debug("Show Incoming Call");
+
+        try {
+            TransactionBuilder builder = performInitialized("incomingCallIcon");
+
+            //Enable call notifications
+            builder.write(ctrlCharacteristic, new byte[] {HPlusConstants.COMMAND_SET_INCOMING_CALL, 1 });
+
+            //Show Call Icon
+            builder.write(ctrlCharacteristic, HPlusConstants.COMMAND_ACTION_INCOMING_CALL);
+
+            //builder = performInitialized("incomingCallText");
+            builder.queue(getQueue());
+
+            try {
+                Thread.sleep(200);
```

Comment:
We try hard to avoid such timing issues. Why do you need to sleep here? Do you have to wait for a response of the device before you can send the next command?

So far we usually do this with "Operations" that are registered as listeners and can react to gatt characteristic change events. For simple "confirmation" responses I'd like to simplify this so that we can avoid creating an operation, but I haven't done this yet.

However, instead of Thread.sleep(), you can add a `WaitAction` to the queue (and thus use the same transaction for the whole stuff).